File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.java
class Solution {
    public List<List<Integer>> findMatrix(int[] nums) {
        int maxFreq = 0;
        int n = nums.length;
        int[] freq = new int[n+1];

        for(int x : nums) {
            freq[x]++;
            maxFreq = Math.max(maxFreq, freq[x]);
        }

        List<List<Integer>> ans = new ArrayList<>(maxFreq);
        for (int i = 0; i < maxFreq; i++) {
            ans.add(new ArrayList<>());
        }

        for(int i = 0; i<freq.length; i++) {
            for(int j = 0; j<freq[i]; j++) {
                ans.get(j).add(i);
            }
        }

        return ans;
    }
}

File: 2610-convert-an-array-into-a-2d-array-with-conditions/2610-convert-an-array-into-a-2d-array-with-conditions.py
class Solution(object):
    def findMatrix(self, nums):
        max_freq = 0
        freq = [0]*(len(nums) + 1)

        for x in nums:
            freq[x] += 1
            max_freq = max(max_freq, freq[x])
        
        ans = [[] for _ in range(max_freq)]
        for i in range(len(freq)):
            for j in range(freq[i]):
                ans[j].append(i)

        return ans

        